source/index.txt

```diff
@@ -1,13 +1,169 @@
 
 .. default-domain:: mongodb
 
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
 Welcome to the MongoDB Commandline Tools Documentation!
 
+Core Processes
+--------------
+
+The core components in the MongoDB tools package are: 
+
+.. list-table::
+  
+   * - :binary:`~bin.mongos`
+
+     - the controller and query router for :term:`sharded clusters
+       <sharded cluster>`
+
+   * - :binary:`~bin.mongo` 
+     - the interactive MongoDB Shell.
+ 
+.. toctree::
+   :maxdepth: 1
+   :hidden:
+
+   /mongos
+   /mongo
+
+Windows Services
+----------------
+
+The :binary:`~bin.mongos.exe` describes the options available for
+configuring the MongoDB query router when running as a Windows Service.
+The :binary:`~bin.mongos.exe` binaries provide a superset of the
+:binary:`~bin.mongos` options.
+
+- :doc:`/mongos.exe`
+
+.. toctree::
+   :maxdepth: 1
+   :hidden:
+
+   /mongos.exe
+
+Binary Import and Export Tools
+------------------------------
+
+:binary:`~bin.mongodump` provides a method for creating :term:`BSON`
+dump files from the :binary:`~bin.mongod` instances, while
+:binary:`~bin.mongorestore` makes it possible to restore these
+dumps. :binary:`~bin.bsondump` converts BSON dump files into
+:term:`JSON`.
+
+- :doc:`/mongodump`
+- :doc:`/mongorestore`
+- :doc:`/bsondump`
+   
+.. toctree::
+   :maxdepth: 1
+   :hidden:
+
+   /mongodump
+   /mongorestore
+   /bsondump
+
+Data Import and Export Tools
+----------------------------
+
+:binary:`~bin.mongoimport` provides a method for taking data in :term:`JSON`,
+:term:`CSV`, or :term:`TSV` and importing it into a :binary:`~bin.mongod`
+instance. :binary:`~bin.mongoexport` provides a method to export data from
+a :binary:`~bin.mongod` instance into JSON, CSV, or TSV.
+
+.. note::
+
+   The conversion between BSON and other formats lacks full
+   type fidelity. Therefore you cannot use :binary:`~bin.mongoimport` and
+   :binary:`~bin.mongoexport` for round-trip import and export operations.
+   
+- :doc:`/mongoimport`
+- :doc:`/mongoexport`
+   
+.. toctree::
+   :maxdepth: 1
+   :hidden:
+
+   /mongoimport
+   /mongoexport
+
+Diagnostic Tools
+----------------
+
+.. note::
+
+   Starting in version 4.0, MongoDB offers :manual:`free Cloud monitoring
+   </administration/free-monitoring>` for standalones and replica sets.
+   For more information, see :manual:`/administration/free-monitoring`.
+
+:binary:`~bin.mongostat` and :binary:`~bin.mongotop`
+provide diagnostic information related to the current operation of a
+:binary:`~bin.mongod` instance.
+
+:binary:`~bin.mongoreplay` is a traffic capture and replay tool for MongoDB
+that you can use to inspect and record commands sent to a MongoDB
+instance, and then replay those commands back onto another host at a
+later time.  :binary:`~bin.mongoreplay` is available for Linux and macOS.
+
+MongoDB 3.4 provides :binary:`~bin.mongoldap` for testing native operating system
```

Comment:
mongoldap isn't included in the commandline tools, to my knowledge. See https://github.com/mongodb/mongo-tools